15-transformer.ipynb

```diff
@@ -354,10 +354,10 @@
     "        [p / np.power(10000, 2 * (j // 2) / dim) for j in range(dim)]\n",
     "        for p in range(nb_p)\n",
     "    ])\n",
-    "    E[:, 0::2] = torch.FloatTensor(np.sin(theta[:, 0::2]))\n",
-    "    E[:, 1::2] = torch.FloatTensor(np.cos(theta[:, 1::2]))\n",
     "    E.detach_()\n",
     "    E.requires_grad = False\n",
+    "    E[:, 0::2] = torch.FloatTensor(np.sin(theta[:, 0::2]))\n",
+    "    E[:, 1::2] = torch.FloatTensor(np.cos(theta[:, 1::2]))\n",
```

Comment:
Why have these lines been moved around?

Reply:
![error](https://user-images.githubusercontent.com/88493484/144694734-7ad625b0-4bf3-441e-9200-f7d7cd997d4e.png)
Not sure about your case, this would give me this error in the model creation block.

Reply:
When generating `position_embeddings = nn.Embedding(max_position_embeddings, d_model)` we should say that we do not require gradients because we don't want to learn the embeddings.